src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -97,7 +97,7 @@ private BoundStatement LowerSwitchStatement(BoundSwitchStatement node)
                 // Evaluate the input and set up sharing for dag temps with user variables
                 BoundDecisionDag decisionDag = ShareTempsIfPossibleAndEvaluateInput(
                     node.GetDecisionDagForLowering(_factory.Compilation),
-                    loweredSwitchGoverningExpression, result, out _);
+                    loweredSwitchGoverningExpression, expr => result.Add(_factory.ExpressionStatement(expr)), out _);
```

Comment:
Could pass the builder directly and wrap afterwards. Every other place use an expression builder instead of statement.